packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts

```diff
@@ -448,6 +448,16 @@ async function askAdditionalAuthQuestions(context, authConfig, defaultAuthType)
 
       authConfig.additionalAuthenticationProviders.push(config);
     }
+  } else {
+    const currentAuthConfig = getAppSyncAuthConfig(context.amplify.getProjectMeta());
```

Comment:
move currentAuthConfig on line 429 out of if block and reuse in if and else

Reply:
fixed